test/unit/domain/users-ips/ip-metadata-validator.spec.ts

```diff
@@ -0,0 +1,154 @@
+import {
+  InvalidIPMetadataASNError,
+  InvalidIPMetadataCountryError,
+  InvalidIPMetadataProxyError,
+  MissingIPMetadataError,
+} from "@domain/errors"
+import { IPMetadataValidator } from "@domain/users-ips/ip-metadata-validator"
+
+const defaultConfig = {
+  denyIPCountries: [],
+  allowIPCountries: [],
+  denyASNs: [],
+  allowASNs: [],
+}
+
+const defaultIpInfo: IPType = {
+  ip: "89.187.173.251" as IpAddress,
+  firstConnection: new Date(),
+  lastConnection: new Date(),
+  asn: "AS60068",
+  provider: "ISP",
+  country: "United States",
+  isoCode: "US",
+  region: "Florida",
+  city: "Miami",
+  proxy: false,
```

Comment:
Could also use a name like `isViaProxy` or something similar to be clearer here

Reply:
I spent some time checking this and probably require a migration to have consistent data.. do you mind if we include this in another PR?